File: app/src/main/java/com/sky/chowder/ui/fragment/RecycleFragment.java
package com.sky.chowder.ui.fragment;

import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import com.sky.base.BaseFragment;
import com.sky.chowder.R;
import com.sky.chowder.ui.adapter.CollapsingAdapter;

import butterknife.BindView;

/**
 * Created by SKY on 2015/11/28 14:52.
 */
public class RecycleFragment extends BaseFragment {
    @BindView(R.id.recycler)
    RecyclerView mRecyclerView;

    @Override
    protected int getLayoutResId() {
        return R.layout.include_recycler;
    }

    @Override
    protected void initPresenter() {

    }

    @Override
    protected void initialize() {

    }

    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        mRecyclerView.setLayoutManager(new LinearLayoutManager(mRecyclerView.getContext()));
        mRecyclerView.setAdapter(new CollapsingAdapter(R.layout.item_recycle_card_main));
    }
}

File: app/src/main/java/com/sky/chowder/ui/fragment/RecycleFragment.kt
package com.sky.chowder.ui.fragment

import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import com.sky.base.BaseFragment
import com.sky.chowder.R
import com.sky.chowder.ui.adapter.CollapsingAdapter
import com.sky.chowder.ui.presenter.SolarP
import kotlinx.android.synthetic.main.include_recycler.*

/**
 * Created by SKY on 2015/11/28 14:52.
 */
class RecycleFragment : BaseFragment<SolarP>() {
    override fun getLayoutResId(): Int = R.layout.include_recycler
    override fun creatPresenter() =SolarP(activity!!)

    override fun initialize(savedInstanceState: Bundle?) {

    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        recycler.layoutManager = LinearLayoutManager(recycler.context)
        recycler.adapter = CollapsingAdapter(R.layout.item_recycle_card_main)
    }
}
